model/LongRunningWorker.php

```diff
@@ -224,4 +223,15 @@ private function getWaitInterval()
             return self::WAIT_INTERVAL;
         }
     }
+
+    private function hasEnoughSpace(): bool
+    {
+        if (!$this->queuer instanceof QueueInterface || $this->queuer->hasPreFetchedMessages()) {
```

Comment:
Based on the constructor, I think the queuer will always be an instance of QueueInterface, no? 

https://github.com/oat-sa/extension-tao-task-queue/pull/169/files#diff-b450c1dd42dc6ba706fa42416c273a4bbf8fb3c9608267e94598bbabeabf1c87R49

```suggestion
        if ($this->queuer->hasPreFetchedMessages()) {
```

Reply:
No. If `--queue` is not provided - [`QueueDispatcherInterface`](https://github.com/oat-sa/extension-tao-task-queue/blob/a9633c0dcef07ac5ba2a1ea7692ba4e87a5eac30/scripts/tools/RunWorker.php#L62) will be used instead of `QueueInterface`.
And `--limit` can be provided only if `--queue` provided 🙂

Reply:
True, this names are confuse 😅 . Do not you think than to avoid breaking open/closed that we should implement the method `hasPreFetchedMessages ` on the `QueuerInterface` instead? As I see this class only accepts instances of `QueuerInterface`